ledgers/ccf/transaction_processor/pdo_tp.cpp

```diff
@@ -116,6 +116,13 @@ namespace ccfapp
             expected_sgx_measurements.mrenclave = in.mrenclave;
             expected_sgx_measurements.basename = in.basename;
             expected_sgx_measurements.ias_public_key = in.ias_public_key;
+            expected_sgx_measurements.sgx_debug_flag = in.sgx_debug_flag;
+
+            // post assignment checks
+            if(expected_sgx_measurements.sgx_debug_flag.length() != 1)
+                return ccf::make_error(
```

Comment:
again puzzled as to why the flag is not a bool? and if even it is a string, why would we check its length ? we have expected flag and actual flag, they need to match, regardless of length.